### PR TITLE
v9 audit Layer#isLoaded implementations

### PR DESCRIPTION
**Pull request by chrisgervang**
v9 port of #8290 

#### Change List
- Wait for `scenegraph` to parse in `ScenegraphLayer`
- Wait for `model` to parse in `SimpleMeshLayer`
- Optional chain in `WMSLayer#isLoaded`

### CODE REVIEW

File: modules/geo-layers/src/terrain-layer/terrain-layer.ts [hunk 1]
```diff
@@ -343,6 +343,10 @@ export default class TerrainLayer<ExtraPropsT extends {} = {}> extends Composite
       );
     }
 
+    if (!elevationData) {
```
Comment (ibgreen) [inline modules/geo-layers/src/terrain-layer/terrain-layer.ts:346]: Nit: This does seem unrelated...

**Review by ibgreen — APPROVED** [review]

**Comment by ibgreen** [comment]
@chrisgervang ready to merge, but seems a conflict has popped up.

File: modules/geo-layers/src/terrain-layer/terrain-layer.ts (hunk 1 above)
Comment (chrisgervang) [reply modules/geo-layers/src/terrain-layer/terrain-layer.ts:346]: https://github.com/visgl/deck.gl/pull/8290#discussion_r1398842208
